scripts/create-k3s-cluster-with-SSL-passthrough.sh

```diff
@@ -81,6 +81,29 @@ fi
 mkdir -p ~/.kube
 export KUBECONFIG=~/.kube/config
 sudo kubectl --kubeconfig=/etc/rancher/k3s/k3s.yaml config view --raw > "$KUBECONFIG"
+
+# Wait for k3s service to be ready and kubectl to work
+echo "Waiting for k3s service to be ready via socket file..."
+timeout=300
+elapsed=0
+while [ $elapsed -lt $timeout ]; do
+    if [ -S /run/k3s/containerd/containerd.sock ] && kubectl get nodes >/dev/null 2>&1; then
```

Comment:
Is `[ -S /run/k3s/containerd/containerd.sock ] &&` necessary here? Can `kubectl get nodes` succeed when that socket does not exist?